sdk/ml/azure-ai-ml/azure/ai/ml/_azure_environments.py

```diff
@@ -189,3 +192,130 @@ def _resource_to_scopes(resource):
     """
     scope = resource + "/.default"
     return [scope]
+
+
+_AZUREML_AUTH_CONFIG_DIR_ENV_NAME = 'AZUREML_AUTH_CONFIG_DIR'
+def _get_config_dir():
+    """Folder path for azureml-core to store authentication config"""
+    _AUTH_FOLDER_PATH = os.path.expanduser(os.path.join('~', '.azureml', "auth"))
+    if os.getenv(_AZUREML_AUTH_CONFIG_DIR_ENV_NAME, None):
+        return os.getenv(_AZUREML_AUTH_CONFIG_DIR_ENV_NAME, None)
+    else:
+        if sys.version_info > (3, 0):
+            os.makedirs(_AUTH_FOLDER_PATH, exist_ok=True)
+        else:
+            if not os.path.exists(_AUTH_FOLDER_PATH):
+                os.makedirs(_AUTH_FOLDER_PATH)
+
+        return _AUTH_FOLDER_PATH
+
+GLOBAL_CONFIG_DIR = _get_config_dir()
+CLOUD_CONFIG_FILE = os.path.join(GLOBAL_CONFIG_DIR, 'clouds.config')
+DEFAULT_TIMEOUT = 30
+_DEFAULT_ARM_URL = "https://management.azure.com/metadata/endpoints?api-version=2019-05-01"
+_ARM_METADATA_URL_ENV_NAME = "ARM_METADATA_URL"
+
+def _get_clouds_by_metadata_url(metadata_url, timeout=DEFAULT_TIMEOUT):
+    """Get all the clouds by the specified metadata url
+
+        :return: list of the clouds
+    """
+    try:
+        import requests
+        module_logger.debug('Start : Loading cloud metatdata from the url specified by {0}'.format(metadata_url))
+        with requests.get(metadata_url, timeout=timeout) as meta_response:
+            arm_cloud_dict = meta_response.json()
+            cli_cloud_dict = _convert_arm_to_cli(arm_cloud_dict)
+            module_logger.debug('Finish : Loading cloud metatdata from the url specified by {0}'.format(metadata_url))
+            return cli_cloud_dict
+    except Exception as ex:  # pylint: disable=broad-except
+        module_logger.warning("Error: Azure ML was unable to load cloud metadata from the url specified by {0}. {1}. "
+                        "This may be due to a misconfiguration of networking controls. Azure Machine Learning Python SDK "
+                        "requires outbound access to Azure Resource Manager. Please contact your networking team to configure "
+                        "outbound access to Azure Resource Manager on both Network Security Group and Firewall. "
+                        "For more details on required configurations, see "
+                        "https://docs.microsoft.com/azure/machine-learning/how-to-access-azureml-behind-firewall.".format(
+            metadata_url, ex))
+
+def _convert_arm_to_cli(arm_cloud_metadata_dict):
+    cli_cloud_metadata_dict = {}
+    for cloud in arm_cloud_metadata_dict:
+        cli_cloud_metadata_dict[cloud['name']] = {
+            EndpointURLS.AZURE_PORTAL_ENDPOINT: cloud["portal"],
```

Comment:
This expects the response to always have these keys. In case it does not it will cause issue for SDK to load. Lets add error handling logic in case the response does not have this data